client/version/latest.go

```diff
@@ -51,32 +52,75 @@ func (f *finder) All() ([]Asset, error) {
 }
 
 func (f *finder) Latest(os, arch string) (Asset, error) {
-	assets, err := f.All()
+	assets, err := f.assetsBy(platform(os, arch))
 	if err != nil {
 		return asset{}, err
 	}
-	var latest Asset
-	for _, asset := range assets {
-		// is right platform
-		if asset.Arch() != arch || asset.Os() != os {
-			continue
-		}
 
-		// first version that matches this platform
-		if latest == nil {
-			latest = asset
-			continue
-		}
+	if len(assets) == 0 {
+		return asset{}, errors.New("no version found")
+	}
 
+	latest := assets[0]
+	for _, asset := range assets {
 		// is this new greater then the one we have
 		if asset.Semver().GT(latest.Semver()) {
 			latest = asset
 		}
 	}
 
+	return latest, nil
+}
+
+func (f *finder) LatestForMajor(os, arch string, major uint64) (Asset, error) {
+	assets, err := f.assetsBy(platformMajor(os, arch, major))
```

Comment:
I think we can reuse the code here by adding sonething like
```
func (f *finder) assets(fn assetFilterFunc) (assets []Asset, latest Asset, err error) {
assets, err = f.assetsBy(fn)
	if err != nil {
		return
	}
 	if len(assets) == 0 {
                err = errors.New("no version found")
		return
	}
        latest = assets[0]
        return
```

Reply:
hmmm. yeah, looks cleaner, although not fan of returning multiple thing (multi-values)